src/fragments/lib-v1/ssr/nextjs/getting-started.mdx

```diff
@@ -0,0 +1,335 @@
+## Prerequisites
+
+If you have not done so already, please follow the [create application](/lib-v1/project-setup/create-application/q/platform/js/)
+guide to create your Next.js project, and Amplify backend.
+
+If you already have an existing Amplify backend, you can run the following command
+at the root of you project to integrate with your existing backend.
+
+```bash
+amplify pull
+```
+
+## Configuration
+
+### Configure Amplify for Server Side Usage
+
+To enable Amplify library API usage on the server side, you will need to update
+your `next.config.js`.
+
+Import the Amplify configuration object from the `amplifyconfiguration.json` file
+that is generated by Amplify CLI, and use the `withAmplify` utility functions to
+update your Next.js app configuration.
+
+For example, the `next.config.js` file may contain the following content.
+
+```javascript
+const { withAmplify } = require('@aws-amplify/adapter-nextjs/with-amplify');
+const config = require('./src/amplifyconfiguration.json');
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  // your Next.js app configuration
+}
+
+module.exports = withAmplify(nextConfig, config);
+```
+
+### Configure Amplify for Client Side Usage
+
+When you use the Amplify library on the client-side of your Next.js app, you will
+need to configure Amplify by calling the Amplify.configure as you would to use
+Amplify in a single-page application.
+
+<Callout>
+
+**NOTE:** To use the Amplify library on the client side in a Next.js app, you will
+need to set `ssr` to `true` when calling `Amplify.configure`.
+
+</Callout>
+
+```typescript
+'use client';
+
+import config from '@/aws-exports';
+import { Amplify } from 'aws-amplify';
+
+Amplify.configure(config, {
+    ssr: true, // required when using Amplify with Next.js
+});
+
+export default function RootLayoutThatConfiguresAmplifyOnTheClient({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return children;
+}
+```
+
+<Callout>
+
+**TIP:** To avoid repetitive calls to `Amplify.configure`, you can call it once
+in a top-level client-side rendered layout component.
+
+</Callout>
+
+## Authentication
+
+You can use the Amplify Auth category APIs to sign up and sign in your end users
+on the client side. With setting `ssr: true` when calling `Amplify.configure`,
+the Amplify library uses cookies to store tokens, which will be sent along with
+HTTP requests to your Next.js app server.
+
+### Manage Auth Session with The `middleware`
```

Comment:
Nit, is `middleware` a more commonly used feature than Page Router or App Router, if not should we move App Router to this section?

Reply:
The middleware is available in both router mechanisms.